cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/PythonAddDeclarationsPass.kt

```diff
@@ -72,6 +72,7 @@ class PythonAddDeclarationsPass(ctx: TranslationContext) : ComponentPass(ctx) {
         when (node) {
             is AssignExpression -> handleAssignExpression(node)
             is ForEachStatement -> handleForEach(node)
+            is Reference -> handleReference(node)
```

Comment:
I am pretty sure I intentionally removed this here a while ago. We only want to handle references that are inside an assign expression. Is it needed?

Reply:
This depends on how we implement `a: int`. We now (after my latest change to this PR) have a `AssignExpression` with an empty list as rhs.
What do we do for `a`? This is currently an `Reference` and should then probably be handled as an `AssignExpression` with empty rhs, too. Once we do this, there is no need for this branch.

Reply:
Well in any other language `a: int` would be just a `VariableDeclaration` or more specifically a `DeclarationStatement` with a `VariableDeclaration`.

As which construct is `a: int` parsed in the python AST? is it parsed as an assignment?

Reply:
> What do we do for `a`? This is currently an `Reference` and should then probably be handled as an `AssignExpression` with empty rhs, too. Once we do this, there is no need for this branch.

Semantically this would also be a declaration in my view - if we compare it with other, more sane languages ;)

Reply:
> Well in any other language `a: int` would be just a `VariableDeclaration` or more specifically a `DeclarationStatement` with a `VariableDeclaration`.
> 
> As which construct is `a: int` parsed in the python AST? is it parsed as an assignment?

It's an `AnnAssign` with no value.

```python
AnnAssign(
      target=Name(id='a', ctx=Store()),
      annotation=Name(id='int', ctx=Load()),
      simple=1)
```

Reply:
Removed the line